### ISSUE TITLE
Cannot View Differences on Github Desktop

### ISSUE THREAD

**Issue by mechPenSketch** [desktop/desktop#14750]
### The problem

On some of my projects, when I try to see differences between a text file, all I got is the following message: 

> This binary file has changed.
> Open file in external program.

This problem is similar to https://github.com/desktop/desktop/issues/7857; however, that issue seems to be from using non-UTF8 characters -- I'm pretty sure I'm not using that in any of my projects.

### Release version

3.0.1 (x64)

### Operating system

Windows 10

### Steps to reproduce the behavior

Tbh, I don't even know how to reproduce this. However, I know which public projects are affected:

* [My JRPG Fragment - Overworld (Dialogic)](https://github.com/mechPenSketch/Jrpg_Fragment-Overworld-Dialogic)
* [Battle System](https://github.com/mechPenSketch/Jrpg_Fragment-TurnBased_Combat)

Maybe you can clone them and see whether the differences are viewable on your side?

### Log files

[2022-06-07.desktop.production.log](https://github.com/desktop/desktop/files/8852904/2022-06-07.desktop.production.log)
Attached above is today's log files that includes where I've looked at a private project that has this issue. The private project is named "Isekai".

I forgot when this started, and the past log has been long deleted.

### Screenshots

![github_differences](https://user-images.githubusercontent.com/15977859/172373969-d4243e7e-beb3-45d1-afba-630bc6261472.png)

### Additional context

I have tried solving this issue on my own, but am at my wit's end. Thus far, I have tried:
* changing text settings in `.gitattribute` to the following combinations:
```
  * text=auto eol=lf
```
```
  * text=auto
  * text eol=lf
```
* Making sure my text files are set to Unix (LF) and UTF-8

I can view the differences on github website, though.

**Comment by steveward** [comment]
@mechPenSketch thanks for the issue. I would try editing your `.gitattributes` with the following line to diff the file:

`*.tscn diff`

Let me know if that does not get things working for you.

**Comment by mechPenSketch** [comment]
Okay, I have added the line and pushed the change, but the past differences is still not displayed.

**Comment by mechPenSketch** [comment]
At first, I thought it might have to do with .gitattribute, so I looked through them from each of the affected projects. Then I read through their history and noticed this:

![github_differences2](https://user-images.githubusercontent.com/15977859/172585880-e84b9c5a-f8ce-4351-967d-1171d9503970.png)

Why is there a space in `* .png` when this doesn't apply to other extensions? I compared it with my unaffected projects and noticed they they don't have such space. So I removed it and the differences become viewable on Github Desktop!

In hindsight, I probably added the space to make the format consistent, not knowing it consequences afterwards.
